core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -925,6 +925,21 @@ private boolean isUnique(Set<ImmutableBitSet> uniqueKeys, ImmutableBitSet key) {
     assertUniqueConsistent(rel);
   }
 
+  @Test public void testFullOuterJoinUniqueness1() {
+    final String sql = "select e.jb, d.nm \n"
+        + "from (select cast(null as int) jb from sales.emp "
+        + " where empno = 10 group by cast(null as int)) as e \n"
+        + "full outer join (select cast (null as int) nm from sales.dept "
+        + "group by cast(null as int)) as d on e.jb = d.nm \n"
+        + "group by e.jb, d.nm";
+    RelNode rel = convertSql(sql);
```

Comment:
I think we should use more readable name such as `empno` and `deptno` instead of `jb` and `nm`.

Reply:
@chunweilei `nm` and `jb` are just alias and not underlying column `empno` or `deptno` (those couldn't be used due to them being non-nullable). Does `alias1/alias2` sound better?

Reply:
What I mean is we can use `empno` as alias like` cast (null as int) empno`.

Reply:
I have updated the pull request to take your feedback into account.

Reply:
I see, makes sense. However, it seems comments are not consistent wrt behavior. Probably there is not a rule that makes a distinction between 'false' and 'null' right now, so it is fine. Maybe you can create a follow-up (you can leave it unassigned) so we can discuss and fix it. I will merge this patch shortly.